csv2/source.go

```diff
@@ -0,0 +1,103 @@
+package csv2
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"strings"
+
+	"github.com/pilosa/pdk"
+	"github.com/pkg/errors"
+)
+
+type Source struct {
+	rs pdk.RawSource
+
+	cur    pdk.NamedReadCloser
+	scan   *bufio.Scanner
+	header []string
+	line   int
+}
+
+func FromRawSource(rs pdk.RawSource) *Source {
```

Comment:
Should this be renamed `NewSourceFromRawSource` to be consistent with the json source?